src/gen/conditions.rs

```diff
@@ -992,6 +994,40 @@ fn test_invalid_spend_list_terminator() {
     );
 }
 
+#[cfg(test)]
+#[rstest]
+#[case(ASSERT_SECONDS_ABSOLUTE, "104")]
+#[case(ASSERT_SECONDS_RELATIVE, "101")]
+#[case(ASSERT_HEIGHT_RELATIVE, "101")]
+#[case(ASSERT_HEIGHT_ABSOLUTE, "100")]
+#[case(RESERVE_FEE, "100")]
+#[case(CREATE_COIN_ANNOUNCEMENT, "{msg1}")]
+#[case(ASSERT_COIN_ANNOUNCEMENT, "{c11}")]
+#[case(CREATE_PUZZLE_ANNOUNCEMENT, "{msg1}")]
+#[case(ASSERT_PUZZLE_ANNOUNCEMENT, "{p21}")]
+#[case(ASSERT_MY_AMOUNT, "123")]
+#[case(ASSERT_MY_COIN_ID, "{coin12}")]
+#[case(ASSERT_MY_PARENT_ID, "{h1}")]
+#[case(ASSERT_MY_PUZZLEHASH, "{h2}")]
+#[case(CREATE_COIN, "{h2} (42 (({h1})")]
+#[case(AGG_SIG_UNSAFE, "{pubkey} ({msg1}")]
+#[case(AGG_SIG_ME, "{pubkey} ({msg1}")]
+fn test_extra_arg_mempool(#[case] condition: ConditionOpcode, #[case] arg: &str) {
+    // extra args are disallowed in mempool mode
+    assert_eq!(
+        cond_test_flag(
+            &format!(
+                "((({{h1}} ({{h2}} (123 ((({} ({} ( 1337 )))))",
```

Comment:
No change necessary, but I'm curious, why didn't you write `{condition as u8}` for the first `{}`? (etc.) Is that legal rust?

Reply:
I'm under the impression that you're not allowed to have arbitrary expression in the string interpolation in rust, just lvalues. I didn't try your suggestion though